### PR TITLE
apinger: improve uci and procd support

### PR DESCRIPTION
**Pull request by jempatel**
### UCI and procd improvements

- convert apinger into procd instances
- generate instance specific apinger.conf from uci
- hotplug handling for apinger alarms
- restart apinger interface instance on ifup action of interface
- don't exit on packet count mismatch, allows to use apinger as monitor
  for multiple targets handling
- add srcip option to target configuration, allows specifying source ip
  used to monitor target
- allow creating status file in script parseable format

Patches are ported against latest version of apinger and referenced from
https://git.pld-linux.org/?p=packages/apinger.git;a=summary

Signed-off-by: Jaymin Patel <jem.patel@gmail.com>

Maintainer: @samm-git
Run tested:  x86-64, aarch64, virtual box, marvell_armada-3720-db, v21.02.3

Description:

Now apinger can be used to manage multiple targets/interfaces from UCI configuration.

packaged uci config (`/etc/config/apinger`) file provides examples with commented sections.

### UCI sections:

**interface**

init script would read all interface sections and it would start an instance of each interface. interface section name must match with `/etc/config/network` interface section, apinger would pick srcip from interface section.

```
config interface 'wan'
	option debug '0'
	option status_interval '5'
```

**alarm_delay**

for each instance, alarm_delay would add a delay alarm

```
# delay in ms
config alarm_delay 'delay200'
	option delay_low '100'
	option delay_high '200'
```

**alarm_loss**

for each instance, alarm_loss would add a loss alarm

```
# loss in %
config alarm_loss 'loss50'
	option percent_low '30'
	option percent_high '50'
```

**alarm_down**

for each instance, alarm_down would add a down alarm

```
# time in seconds
config alarm_down 'down1s'
	option time '1'
```

**target**

target to monitor

```
config target 'target1'
	option interface 'wan'
	option address '8.8.8.8'
	option alarm_delay '<alarm_name>'
	option alarm_loss '<alarm_name>'
	option alarm_down '<alarm_down>'
        option avg_delay_samples '50'
        option avg_loss_samples '10'
        option avg_loss_delay_samples '10'
	option probe_interval '5'
```

address and interface are required options for the target, all other config options would pick default values from the default config.

### Hotplug

On apinger alarm, It would trigger a hotplug action with a few variables,

```
ACTION=up|down # alarm goes up or down
INSTANCE=instance_name (interface name) # alarm generated from this instance
APINGER_TARGET
APINGER_SRCIP
APINGER_TARGET_ID=<uci cfg id of target>
APINGER_ALARM=<alarm name>
APINGER_ALARM_TYPE=<alarm_type> # loss/down/delay
APINGER_ALARM_MESSAGE
APINGER_PROBES_SENT
APINGER_PROBES_RECEIVED
APINGER_LOSS # packet loss
APINGER_DELAY # latency/delay
APINGER_TIMESTAMP # time
```

### CODE REVIEW

**Comment by samm-git** [comment]
amazing work, looks good to me

File: net/apinger/Makefile [hunk 1]
```diff
@@ -56,6 +56,14 @@ define Package/apinger/install
 	$(INSTALL_DATA) $(PKG_BUILD_DIR)/src/apinger.conf $(1)/etc/apinger.conf
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/apinger.init $(1)/etc/init.d/apinger
+	$(INSTALL_DIR) $(1)/etc/config
+	$(INSTALL_BIN) ./files/apinger.config $(1)/etc/config/apinger
```
Comment (feckert) [inline net/apinger/Makefile:60]: That is not a INSTALL_BIN please use INSTALL_DATA

File: net/apinger/Makefile [hunk 2]
```diff
@@ -56,6 +56,14 @@ define Package/apinger/install
...
+	$(INSTALL_DIR) $(1)/usr/libexec
+	$(INSTALL_BIN) ./files/apinger-hotplug $(1)/usr/libexec/apinger-hotplug
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/apinger
+	$(INSTALL_BIN) ./files/user.hotplug $(1)/etc/hotplug.d/apinger/01-user
```
Comment (feckert) [inline net/apinger/Makefile:64]: That is not a INSTALL_BIN please use INSTALL_DATA

File: net/apinger/Makefile [hunk 3]
```diff
@@ -56,6 +56,14 @@ define Package/apinger/install
...
+	$(INSTALL_DIR) $(1)/etc/hotplug.d/iface
+	$(INSTALL_BIN) ./files/iface.hotplug $(1)/etc/hotplug.d/iface/25-apinger
```
Comment (feckert) [inline net/apinger/Makefile:66]: That is not a INSTALL_BIN please use INSTALL_DATA

File: net/apinger/Makefile [hunk 4]
```diff
@@ -46,7 +46,7 @@ define Package/apinger/description
 endef
 
 define Package/apinger/conffiles
-/etc/apinger.conf
+/etc/config/apinger
```
Comment (feckert) [inline net/apinger/Makefile:49]: Should we not add also the file `/etc/apinger.user` to conffiles? To preserve this file over a sysupgrade

**Review by feckert — CHANGES_REQUESTED** [review]
Why do you rebase the patchset. Was this not done before correctly?
I am asking, because you did not update the version.

File: net/apinger/Makefile (hunk 1 above)
Comment (jempatel) [reply net/apinger/Makefile]: Updated

File: net/apinger/Makefile (hunk 3 above)
Comment (jempatel) [reply net/apinger/Makefile]: Updated

File: net/apinger/Makefile (hunk 4 above)
Comment (jempatel) [reply net/apinger/Makefile:49]: Yes, added /etc/apinger.usr to conffiles

**Comment by jempatel** [comment]
> Why do you rebase the patchset. Was this not done before correctly? I am asking, because you did not update the version.

Actually, there are no additional changes from the upstream version, new patches were ported on top of the current version.

**Review by feckert — APPROVED** [review]

**Comment by feckert** [comment]
Thanks for your work!

**Pull request merged**